llama_index/vector_stores/zep.py

```diff
@@ -0,0 +1,337 @@
+    ) -> List[str]:
+        """Asynchronously add a list of embedding results to the collection.
+
+        Args:
+            embedding_results (List[NodeWithEmbedding]): Embedding results to add.
+
+        Returns:
+            List[str]: List of IDs of the added documents.
+        """
+        from zep_python.document import DocumentCollection
+
+        if not isinstance(self._collection, DocumentCollection):
+            raise ValueError("Collection not initialized")
+
+        if self._collection.is_auto_embedded:
+            raise ValueError("Collection is auto embedded, cannot add embeddings")
+
+        docs, ids = self._prepare_documents(embedding_results)
+
+        await self._collection.aadd_documents(docs)
+
+        return ids
+
+    def delete(
+        self, ref_doc_id: Optional[str] = None, **delete_kwargs: Any
+    ) -> None:  # type: ignore
+        """Delete a document from the collection.
+
+        Args:
+            ref_doc_id (Optional[str]): ID of the document to delete.
+                Not currently supported.
+            delete_kwargs: Must contain "uuid" key with UUID of the document to delete.
+        """
+        from zep_python.document import DocumentCollection
+
+        if not isinstance(self._collection, DocumentCollection):
+            raise ValueError("Collection not initialized")
+
+        if ref_doc_id and len(ref_doc_id) > 0:
+            raise NotImplementedError(
+                "Delete by ref_doc_id not yet implemented for Zep."
+            )
+
+        if "uuid" in delete_kwargs:
+            self._collection.delete_document(uuid=delete_kwargs["uuid"])
+        else:
+            raise ValueError("uuid must be specified")
+
+    async def adelete(
+        self, ref_doc_id: Optional[str] = None, **delete_kwargs: Any
+    ) -> None:  # type: ignore
+        """Asynchronously delete a document from the collection.
+
+        Args:
+            ref_doc_id (Optional[str]): ID of the document to delete.
+                Not currently supported.
+            delete_kwargs: Must contain "uuid" key with UUID of the document to delete.
+        """
+        from zep_python.document import DocumentCollection
+
+        if not isinstance(self._collection, DocumentCollection):
+            raise ValueError("Collection not initialized")
+
+        if ref_doc_id and len(ref_doc_id) > 0:
+            raise NotImplementedError(
+                "Delete by ref_doc_id not yet implemented for Zep."
+            )
+
+        if "uuid" in delete_kwargs:
+            await self._collection.adelete_document(uuid=delete_kwargs["uuid"])
+        else:
+            raise ValueError("uuid must be specified")
+
+    def _parse_query_result(
+        self, results: List["ZepDocument"]
+    ) -> VectorStoreQueryResult:
+        similarities: List[float] = []
+        ids: List[str] = []
+        nodes: List[TextNode] = []
+
+        for d in results:
+            node = TextNode(text=d.content, metadata=d.metadata)
```

Comment:
Actually, this is not quite correct

We should use the metadata dict to node util function here. I can patch that in

Reply:
Thank you!

Reply:
Fixed!